### DISCUSSION TITLE
How to integrate S3 syncing for embedded linux application

### DISCUSSION THREAD

**Discussion by eden-parallel** [aws4embeddedlinux/meta-aws#264] [Q&A]
Hi!

I have a system that needs to automatically sync data with an S3 bucket. I've been using AWS-CLI for this and was hoping to add it to our system but am wondering if there is a better way to do this? We also use the AWS-CLI to authorize direct calls to AWS services using AWS IoT Core credential provider as described [here](https://docs.aws.amazon.com/iot/latest/developerguide/authorizing-direct-aws.html).

Our Yocto release is Gatesgarth, and it's not included, so I am unsure of how to either get AWS-CLI on my build, or if there's another way to accomplish my goals.

-Eden

**Comment by nateglims** [comment]
Hi Eden,

Sorry about the late reply, we dropped the ball on our end.

We generally recommend using Device identity and credential management instead of direct AWS credentials and the CLI in embedded systems. The link you provided outlines a way to do that, and there is also [this blog post](https://aws.amazon.com/blogs/iot/securely-ingesting-large-sized-payloads-from-iot-devices-to-the-aws-cloud/) with some more concrete details.

**Comment by dwalkes** [comment]
I found this difficult to understand based on existing documentation but put some instructions together in https://github.com/Trellis-Logic/meta-aws-iot-demo/wiki/Creating-IAM-Policies-for-S3-bucket-access and demonstrated the functionality with the project at https://github.com/Trellis-Logic/meta-aws-iot-demo in case that's useful for anyone.
